ops/model.py

```diff
@@ -281,6 +281,20 @@ def get_secret(self, *, id: Optional[str] = None, label: Optional[str] = None) -
         content = self._backend.secret_get(id=id, label=label)
         return Secret(self._backend, id=id, label=label, content=content)
 
+    def get_cloud_spec(self) -> 'CloudSpec':
+        """Get cloud credential information.
+
+        Access the cloud credential information and return the cloud specification
+        used by the model.
+
+        Note: It does not work in a "CaaS" model (Container-as-a-Service,
+        i.e., containerized environment, Kubernetes charm).
```

Comment:
I know the Juju error refers to "CaaS" but from what I can see there's very little use of this term in the Juju docs. I'm not sure what the state is here - is this a term that's being adopted, or an old one that's going away, or just one that is only occasionally used? If @benhoyt doesn't know, we could ask the Juju team.

It seems like a simpler version using the names we generally use might work better, like:

```suggestion
        Note: This information is only available for machine charms, not Kubernetes sidecar charms.
```

Similarly in the "Raises" section, I would say something like "Kubernetes model" rather than "CaaS".